x-pack/plugins/ml/server/models/model_management/models_provider.ts

```diff
@@ -494,14 +495,19 @@ export class ModelsProvider {
   }
 
   /**
-   * Provides an ELSER model name and configuration for download based on the current cluster architecture.
-   * The current default version is 2. If running on Cloud it returns the Linux x86_64 optimized version.
-   * If any of the ML nodes run a different OS rather than Linux, or the CPU architecture isn't x86_64,
-   * a portable version of the model is returned.
+   * Provides an appropriate model ID and configuration for download based on the current cluster architecture.
+   *
+   * @param modelName
+   * @param options
+   * @returns
    */
-  async getELSER(options?: GetElserOptions): Promise<ModelDefinitionResponse> | never {
-    const modelDownloadConfig = await this.getModelDownloads();
-
+  async getCuratedModelConfig(
+    modelName: ElasticCurateModelName,
+    options?: GetModelDownloadConfigOptions
+  ): Promise<ModelDefinitionResponse> | never {
+    const modelDownloadConfig = (await this.getModelDownloads()).filter(
+      (model) => model.modelName === modelName
```

Comment:
I find the prop `name` in `ModelDefinitionResponse` confusing, is it worth changing it to `id` to explicitly differentiate it from `modelName` ?
Doesn't have to be done in this PR but I think it is worth changing. I just remembered how confused I was when I added the function `installElasticModel` to this file.